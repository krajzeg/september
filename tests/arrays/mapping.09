array := [1, 2, 3]
doubled := array.map |x| { x * 2 }

print("If [1,2,3].map() works, this should be 2, 4, 6:", doubled.toString())
print("The length after mapping should still be 3:", doubled.length())

neverCalled := |x| { print("But the mapping was called.") }
print("An empty array mapped should still be empty, and the mapping should never be called:", [].map(neverCalled))
