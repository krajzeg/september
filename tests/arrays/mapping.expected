If [1,2,3].map() works, this should be 2, 4, 6: 2, 4, 6
The length after mapping should still be 3: 3
An empty array mapped should still be empty, and the mapping should never be called: 
